src/link/Elf.zig

```diff
@@ -890,7 +890,7 @@ pub fn flushModule(self: *Elf, comp: *Compilation) !void {
             @panic("TODO: handle .debug_info header exceeding its padding");
         }
         const jmp_amt = first_dbg_info_decl.dbg_info_off - di_buf.items.len;
-        try self.pwriteDbgInfoNops(0, di_buf.items, jmp_amt, false, debug_info_sect.sh_offset);
+        try self.pwriteDbgInfoNops(0, di_buf.items, jmp_amt, false, @intCast(usize, debug_info_sect.sh_offset));
```

Comment:
`pwriteDbgInfoNops` should take an `u64` offset, `usize` is for objects stored in memory.

Reply:
Okay, but the cast still has to happen somewhere, because the `pwritevAll` syscall takes the offset as a usize. So where should this cast happen? Similar question for `copyRangeAll`

Reply:
You have to recursively fix all the offsets using `usize` type, `pwritevAll` is wrong but `os.pwrite` correctly uses a `u64`